docs/design/1533-host-specific-tasklist.md

```diff
@@ -41,32 +41,40 @@ The following sections will first go through the APIs exposed to users, explain
 There will be four new APIs available when writing a workflow:
 
 ```go
-sessionCtx, err := workflow.**CreateSession**(ctx)
+sessionCtx, err := workflow.**CreateSession**(ctx Context, so *SessionOptions)
 ```
 
-A user calls this API to create a session and use the returned sessionCtx to execute activities. All activities executed within the returned sessionCtx are considered to be part of the session and will be executed on the same worker. It’s possible that this call will fail due to various reasons, for example, no worker is available at this time.
+A user calls this API to create a session on the worker that polls the task list specified in the ActivityOptions (or in the StartWorkflowOptions if the task list name is not specified in the ActivityOptions). All activities executed within the returned sessionCtx (a new context which contains metadata information of the created session) are considered to be part of the session and will be executed on the same worker. The sessionCtx will be cancelled if the worker executing this session dies or CompleteSession() is called.
 
-The session will be created on one of the workers that polls from the tasklist specified in the context passed into the CreateSession API. As a result, users need to make sure they are using the correct task list.
+The SessionOptions struct contains two fields: `ExecutionTimeout`, which specifies the maximum amount of time the session can run and `CreationTimeout`, which specifies how long session creation can take before returning an error.
 
-When executing an activity within a session, a user may get two types of errors. The first one is returned from user activities. The session will not be marked as failed in this case, so the user can return whatever error they want and apply their business logic as necessary. If a user wants to end a session due to the error returned from the activity, use the CompleteSession() API below. The second type of error is a special error which says the session has failed due to worker failure and the session is marked as failed in the background. In this case, no activities can be executed using this context. The user can choose how to handle the failure. They can create a new session to retry or end the workflow with an error.
+CreateSession() will return an error if the context passed in already contains an open session. If all the workers are currently busy and unable to handle new sessions, the framework will keep retrying until the CreationTimeout you specified in the SessionOptions has passed before returning an error.
+
+When executing an activity within a session, a user may get three types of errors:
```

Comment:
This document is project proposal, not user documentation. So I think it should use third person(user) instead of second person(you)?

Reply:
Okay, I've updated my remaining comments to reflect the use of third person. In this case, it should be: 
a user may -> a user might

Reply:
It's a fair point. @purplesusan, I think it's best to limit your review of these design docs (anything under docs/design folder) to just catch if anything doesn't belong in public documentation.

Two points about 3rd person though: 

1. "User" is ambiguous. "Developer" is also unclear because it doesn't clarify if that's someone using the Cadence client or someone contributing to Cadence. There's also a sentence below: "the user needs to ensure that the resource is owned by only one worker on a single host." Is that a developer or an ops person?
2. You're probably going to copy this into the public documentation anyway.

Reply:
I think user here refers to whoever is using the session API. Maybe developer is more accurate.

The "user" in the sentence "the user needs to ensure that the resource is owned by only one worker on a single host." is a developer (who configs their worker). Also, right now the ResouceID field is hidden from user as session reestablishing is not implemented.